### PR TITLE
fix(commander): issue with namespace not found

### PR DESCRIPTION
**Pull request by alan-agius4**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [] Increase the version number in the header if appropriate.
- [X] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dslint/dt.json" }`.

### CODE REVIEW

**Comment by alan-agius4** [comment]
👍

**Comment by alan-agius4** [comment]
Can we have this merged please?

File: types/commander/index.d.ts [hunk 1]
```diff
@@ -17,7 +16,7 @@ declare namespace commander {
         unknown: string[];
     }
 
-    class Option {
+    interface Option {
```
Comment (mhegazy) [inline types/commander/index.d.ts]: this does not look right. `Option` seems to be a class, and not an object that is callable with `new`:

```sh
> var commander = require("commander");
undefined
> commander.Option
[Function: Option]
> new commander.Option("--f");
Option {
  flags: '--f',
  required: 0,
  optional: 0,
  bool: true,
  long: '--f',
  description: '' }
> new (new commander.Option("--f"))("--f");
TypeError: (intermediate value) is not a constructor
    at repl:1:1
    at realRunInThisContextScript (vm.js:22:35)
    at sigintHandlersWrap (vm.js:98:12)
    at ContextifyScript.Script.runInThisContext (vm.js:24:12)
    at REPLServer.defaultEval (repl.js:346:29)
    at bound (domain.js:280:14)
    at REPLServer.runBound [as eval] (domain.js:293:12)
    at REPLServer.onLine (repl.js:544:10)
    at emitOne (events.js:101:20)
    at REPLServer.emit (events.js:188:7)
```

**Review by mhegazy — CHANGES_REQUESTED** [review]

File: types/commander/index.d.ts (hunk 1 above)
Comment (alan-agius4) [inline types/commander/index.d.ts]: How are you suggesting to change it? If I can it to a class. I'll get
```
[ts] Duplicate identifier "commander"
```

And if I do the class outside the `namespace`,  and export as

```
declare const commander: commander.StaticCommander & Option;
```

I can still do 

```
new(new program.Option("--f"))("--f")
```

File: types/commander/index.d.ts [hunk 2]
```diff
@@ -5,278 +5,300 @@
...
         /**
-         * Get the name of the command.
+         * Initialize a new `Option` with the given `flags` and `description`.
          *
-         * @return {string}
+         * @param {string} flags
+         * @param {string} [description]
          */
-        name(): string;
+        Command: new (name?: string) => Command;
```
Comment (mhegazy) [inline types/commander/index.d.ts]: `Command: typeof Command`

File: types/commander/index.d.ts [hunk 3]
```diff
@@ -5,278 +5,300 @@
...
 
         /**
-         * Output help information for this command.
+         * Initialize a new `Command`.
          *
-         * @param {() => void} [cb] Callback method
+         * @param {string} [name]
          */
-        outputHelp(cb?: () => void): void;
+        Option: new (flags: string, description?: string) => Option;
```
Comment (mhegazy) [inline types/commander/index.d.ts]: `Option: typeof Option`

Comment (alan-agius4) [inline types/commander/index.d.ts]: thank you for the review @mhegazy, updated as suggested

**Comment by alan-agius4** [comment]
Any update please :)

**Comment by alan-agius4** [comment]
@mhegazy any update please?

**Comment by alan-agius4** [comment]
@mhegazy can we have this merged pls?

**Comment by sanex3339** [comment]
it's happening!

**Pull request merged**